sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/CBDE/MlirExporter.cs

```diff
@@ -137,6 +139,19 @@ public void ExportFunction(MethodDeclarationSyntax method)
                 writer.WriteLine();
                 return;
             }
+
+            IControlFlowGraph cfg;
+            try
+            {
+                cfg = CSharpControlFlowGraph.Create(method.Body, semanticModel);
+            }
+            catch (Exception ex) when (semanticModel.Compilation.GetLanguageVersion() > supportedVersion && !Debugger.IsAttached)
+            {
+                // We don't want to fail when exception occurs for unsupported language version
+                writer.WriteLine($"// Skipping function {method.Identifier.ValueText}{GetAnonymousArgumentsString(method)}, failed to generate CFG: {ex.Message}");
```

Comment:
does this get written in the MLIR?
can you add an integration test to verify this? 

I know it's a bunch of work, but I don't feel comfortable with what we currently have

Reply:
Note from offline discussion: We need a syntax that will cause CFG generation to fail. We have it now, but we will not have it in a long run.
Since it only generates a comment (as the statement above), this should be fine without integration test.